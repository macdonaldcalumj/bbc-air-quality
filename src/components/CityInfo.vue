<template>
  <div class="columns">
    <div class="column">
      <section class="hero is-warning">
        <div class="hero-body">
          <p class="title">
            {{ title }}
          </p>
          <b-autocomplete
            placeholder="e.g. Delhi"
            v-model="cityName"
            :open-on-focus="true"
            :data="filteredDataObj"
            field="name"
            @select="option => (selectedCity = option)"
          >
          </b-autocomplete>
        </div>
      </section>
    </div>
    <div class="column">
      <section class="hero is-success">
        <div class="hero-body">
          <p class="subtitle">
            {{ $t("message.totalCigarettes") }}
          </p>
          <p v-if="selectedCity" class="title">
            {{ selectedCity.cig }}
          </p>
        </div>
      </section>
    </div>
    <div class="column">
      <section class="hero is-danger">
        <div class="hero-body">
          <p class="subtitle">
            AQI
          </p>
          <p v-if="selectedCity" class="title">
            {{ selectedCity.aqi }}
          </p>
        </div>
      </section>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      cityName: '',
      selectedCity: null,
    }
  },
  props: {
    title: {
      default: ''
    },
    cities: {
      default() {
        return []
      }
    }
  },
  computed: {
    filteredDataObj() {
      return this.cities.filter(option => {
        return (
          option.name
            .toString()
            .toLowerCase()
            .indexOf(this.cityName.toLowerCase()) >= 0
        )
      })
    }
  }
}
</script>

<style>

</style>
