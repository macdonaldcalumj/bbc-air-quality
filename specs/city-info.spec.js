import { mount } from "@vue/test-utils";
import CityInfo from "@/components/CityInfo";

test("test mounting component", () => {
  const wrapper = mount(
    CityInfo, {
      mocks: {
      $t: () => {}
    }
  });

  expect(wrapper).toMatchSnapshot()
})

test("test selecting city", async () => {
  const cities = [
    {
      name: "A",
      aqi: "91 PM2.5",
      cig: 4,
    },
    {
      name: "B",
      aqi: "61 PM2.5",
      cig: 3
    },
  ];

  const wrapper = mount(CityInfo, {
    propsData: {
      title: "Select your city",
      cities: cities,
    },
    mocks: {
      $t: () => {},
    },
  });

  expect(wrapper).toMatchSnapshot();
  
  await wrapper.setData({ selectedCity: cities[0] })

  expect(wrapper).toMatchSnapshot()
});
