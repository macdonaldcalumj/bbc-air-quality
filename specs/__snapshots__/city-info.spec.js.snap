// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test mounting component 1`] = `
<div class="columns">
  <div class="column">
    <section class="hero is-warning">
      <div class="hero-body">
        <p class="title">

        </p>
        <b-autocomplete placeholder="e.g. Anne" open-on-focus="true" data="" field="name"></b-autocomplete>
      </div>
    </section>
  </div>
  <div class="column">
    <section class="hero is-success">
      <div class="hero-body">
        <p class="subtitle">

        </p>
        <!---->
      </div>
    </section>
  </div>
  <div class="column">
    <section class="hero is-danger">
      <div class="hero-body">
        <p class="subtitle">
          AQI
        </p>
        <!---->
      </div>
    </section>
  </div>
</div>
`;

exports[`test selecting city 1`] = `
<div class="columns">
  <div class="column">
    <section class="hero is-warning">
      <div class="hero-body">
        <p class="title">
          Select your city
        </p>
        <b-autocomplete placeholder="e.g. Anne" open-on-focus="true" data="[object Object],[object Object]" field="name"></b-autocomplete>
      </div>
    </section>
  </div>
  <div class="column">
    <section class="hero is-success">
      <div class="hero-body">
        <p class="subtitle">

        </p>
        <!---->
      </div>
    </section>
  </div>
  <div class="column">
    <section class="hero is-danger">
      <div class="hero-body">
        <p class="subtitle">
          AQI
        </p>
        <!---->
      </div>
    </section>
  </div>
</div>
`;

exports[`test selecting city 2`] = `
<div class="columns">
  <div class="column">
    <section class="hero is-warning">
      <div class="hero-body">
        <p class="title">
          Select your city
        </p>
        <b-autocomplete placeholder="e.g. Anne" open-on-focus="true" data="[object Object],[object Object]" field="name"></b-autocomplete>
      </div>
    </section>
  </div>
  <div class="column">
    <section class="hero is-success">
      <div class="hero-body">
        <p class="subtitle">

        </p>
        <p class="title">
          4
        </p>
      </div>
    </section>
  </div>
  <div class="column">
    <section class="hero is-danger">
      <div class="hero-body">
        <p class="subtitle">
          AQI
        </p>
        <p class="title">
          91 PM2.5
        </p>
      </div>
    </section>
  </div>
</div>
`;
